build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/release/AddVersionTask.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.release;
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter;
+import com.google.common.annotations.VisibleForTesting;
+
+import org.elasticsearch.gradle.Version;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.initialization.layout.BuildLayout;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+public class AddVersionTask extends AbstractVersionTask {
+    private static final Logger LOGGER = Logging.getLogger(AddVersionTask.class);
+
+    private Version version;
+    private boolean setCurrent;
+
+    @Inject
+    public AddVersionTask(BuildLayout layout) {
+        super(layout);
+    }
+
+    @Option(option = "version", description = "Specifies the version to add")
+    public void version(String version) {
+        this.version = Version.fromString(version);
+    }
+
+    @Option(option = "update-current", description = "Update the 'current' constant to the new version")
+    public void setCurrent(boolean setCurrent) {
+        this.setCurrent = setCurrent;
+    }
+
+    @TaskAction
+    public void executeTask() throws IOException {
+        if (version == null) {
+            throw new IllegalStateException("version has not been specified");
+        }
+
+        Path versionJava = rootDir.resolve(VERSION_PATH);
```

Comment:
Nit: I was confused by this name, I had to look at what was in VERSION_PATH. Maybe naming it VERSION_SRC_FILE_PATH, or renaming versionJava to resolvedVersionFilePath (or something similar)